fs/readdircount_linux_amd64.go

```diff
@@ -0,0 +1,67 @@
+// +build linux,amd64
+
+package fs
+
+import (
+	"errors"
+	"os"
+	"syscall"
+	"unsafe"
+)
+
+func countDirEntries(buf []byte, n int) int {
+	count := 0
+	buf = buf[:n]
+	for len(buf) > 0 {
+		// see man page getdents(2) for struct linux_dirent64
+		reclen := *(*uint16)(unsafe.Pointer(&buf[16]))
```

Comment:
Please use [`unsafe.Offsetof(Dirent{}.Reclen)`](https://github.com/golang/sys/blob/493114f68206f85e7e333beccfabc11e98cba8dd/unix/syscall_linux.go#L1002)  instead

IN fact,